madprops/serializers.py

```diff
@@ -52,7 +54,12 @@ def to_representation(self, value):
         return result_dict
 
     def to_internal_value(self, data):
-        data_list = [{name: value} for (name, value) in data.items()]
+        try:
+            data_list = [{name: value} for (name, value) in data.items()]
+        except AttributeError:
+            raise ValidationError(
+                _('`%(data)s` is not a key value mapping') % {'data': data})
```

Comment:
And since the caller knows what they pass, we could probably simply say `Properties should be a key/value mapping` (without responding with data which might be big enough)